tools/azure-pipelines/jobs-template.yml

```diff
@@ -243,3 +278,9 @@ jobs:
       inputs:
         targetPath: $(DEBUG_FILES_OUTPUT_DIR)
         artifact: logs-${{parameters.stage_name}}-$(DEBUG_FILES_NAME)
+    - script: |
+        mkdir -p $(TESTCONTAINER_CACHE_FOLDER)
+        docker save $(docker image ls --format "{{.Repository}}:{{.Tag}}" | grep -E -- "testcontainers|kafka|elasticsearch|postgres|mysql") -o $(TESTCONTAINER_CACHE_FOLDER)/cache.tar || true
```

Comment:
can we move this into a script? It's likely that the set of cached images will increase of timer, and having to adjust it in 2 places is a recipe for inconsistencies.

Reply:
@zentol I was thinking: perhaps the set of cached images could be part of the cache key? Since if we change that, we'll want to invalidate the cache in any case.

Reply:
_Do_ we want to invalidate it? We are likely just extending the list, so anything in the cache at that time is probably still relevant. We'd just extend it.

Reply:
That said, we do need some mechanism to eventually invalidate (at least part of) the cache, otherwise it should keep accumulating various versions of the same image (e.g., all tags for a particular image released over time)

Reply:
That's the thing - once an entry for a cache key already exists, the contents are not updated on subsequent cache hits. So when we extend the list, the new images won't be persisted to the already existing cache entry. See: https://github.com/microsoft/azure-pipelines-tasks/issues/11600

Reply:
hmm...yes then it makes sense to have it as part of the key. But this does also mean that if we fetch an image a _different_  elasticsearch version that it will never be cached.
Maybe we have to setup a convention such that all images we use must be listed in a specific .java file (/.sh file for the e2e tests), which we then use as a cache key.

Reply:
Yeah - this is why I thought maybe caching based on the contents of pom.xml might be a good fit, since the specific testcontainer container versions are defined in there, for example with elasticsearch 7.5.1 https://github.com/apache/flink/blob/1c67df838ad045484f96a5839c324771285d9cf7/flink-connectors/flink-connector-elasticsearch7/pom.xml#L92

Reply:
@zentol Yeah, this is why I thought maybe matching on pom.xml might be a good idea since we specify the version of the testcontainers to use within them, for example with elasticsearch 7.5.1: https://github.com/apache/flink/blob/1c67df838ad045484f96a5839c324771285d9cf7/flink-connectors/flink-connector-elasticsearch7/pom.xml#L40

Reply:
that's not necessarily the case for other tests though.

Reply:
I won't object to using the pom files, but it's not a long-term solution.